### PR TITLE
Make files visible when searching in ``FileBrowser`` 

### LINKED ISSUE #4214: Searching in the sidebar shows the node that contains matching entries
In the Sidebar, if I search for a string, LMMS shows the node that contains matching entries.

For example, in **My projects** node, there is **Editing_note_volumes.mmp** entry.
This is visible when I expand the node (by clicking on the triangle on its left).

Now I simulate a search, by entering "note" in the search field. 
LMMS starts shortening the nodes' list as I type.
Finally the **My projects** node remains.

But the mmp file is not displayed until I expand the node.
image

image

**Desired:** Please shortlist the matching items without the nodes structure.

---

**Edited by @tresf**
Feedback from @foolo (#4609) and also in alignment with @musikBear's feedback below:
> I still want to see the directories, but expanded.

Expanded folders need to be in this design.

### PR DESCRIPTION
**Pull request by MrTopom**
Try to make files visible when searching in FileBrowser and restore the previous state if search is cancelled

Fixes #4214

### CODE REVIEW

**Review by DomClark — COMMENTED** [review]
If you write `Fixes #4214` in the pull request description (not the title), GitHub will automatically link the PR to the issue so that merging the PR will close the issue. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue.

Also, it's good to try and provide a title that is meaningful in and of itself, as it's used as the commit message when the PR is merged. Right now, someone seeing "Issue#4214" has to go and look up on the internet what the commit does, which can be inconvenient.

File: src/gui/FileBrowser.cpp [hunk 1]
```diff
@@ -201,15 +233,14 @@ bool FileBrowser::filterItems( const QString & filter, QTreeWidgetItem * item )
 
 void FileBrowser::reloadTree()
 {
-	QList<QString> expandedDirs = m_fileBrowserTreeWidget->expandedDirs();
```
[the remark this answers is not in the document]
Comment (MrTopom) [reply src/gui/FileBrowser.cpp:204]: Saving is done only if we start a search.

In the reloadTree function, at the end, depending on the fact there is or not a search filter it should manage the right way.

[the remark this answers is not in the document]
Comment (DomClark) [reply src/gui/FileBrowser.cpp:204]: There's no point restoring the state if it hasn't been saved. Suppose the user has never searched for anything, but decides to refresh the tree. Their previous tree state should still be there.

[the remark this answers is not in the document]
Comment (MrTopom) [reply src/gui/FileBrowser.cpp:204]: Should be corrected

[the remark this answers is not in the document]
Comment (DomClark) [reply src/gui/FileBrowser.cpp:204]: Looks good to me.

**Comment by zonkmachine** [comment]
LGTM! :+1:

**Review by zonkmachine — APPROVED** [review]
Based on testing, approved. Code looks fine too but I'm not a coder by any stretch of the imagination ;).

**Comment by zonkmachine** [comment]
Please, whoever merges this, edit the commit message to be more descriptive.
I suggest just the string from the PR description above "Try to make files visible when searching in FileBrowser and restore the previous state if search is cancelled"

**Review by Veratil — APPROVED** [review]

**Pull request merged**